ext/geos_c_impl/factory.c

```diff
@@ -569,48 +542,19 @@ static VALUE alloc_geometry(VALUE klass)
 /**** INITIALIZATION FUNCTION ****/
 
 
-RGeo_Globals* rgeo_init_geos_factory()
+void rgeo_init_geos_factory()
 {
-  RGeo_Globals* globals;
-  VALUE rgeo_module;
   VALUE geos_factory_class;
-  VALUE wrapped_globals;
-  VALUE feature_module;
-
-  rgeo_module = rb_define_module("RGeo");
-
-  globals = ALLOC(RGeo_Globals);
-
-  // Cache some modules so we don't have to look them up by name every time
-  feature_module = rb_define_module_under(rgeo_module, "Feature");
-  globals->feature_module = feature_module;
-  globals->geos_module = rb_define_module_under(rgeo_module, "Geos");
-  globals->feature_geometry = rb_const_get_at(feature_module, rb_intern("Geometry"));
-  globals->feature_point = rb_const_get_at(feature_module, rb_intern("Point"));
-  globals->feature_line_string = rb_const_get_at(feature_module, rb_intern("LineString"));
-  globals->feature_linear_ring = rb_const_get_at(feature_module, rb_intern("LinearRing"));
-  globals->feature_line = rb_const_get_at(feature_module, rb_intern("Line"));
-  globals->feature_polygon = rb_const_get_at(feature_module, rb_intern("Polygon"));
-  globals->feature_geometry_collection = rb_const_get_at(feature_module, rb_intern("GeometryCollection"));
-  globals->feature_multi_point = rb_const_get_at(feature_module, rb_intern("MultiPoint"));
-  globals->feature_multi_line_string = rb_const_get_at(feature_module, rb_intern("MultiLineString"));
-  globals->feature_multi_polygon = rb_const_get_at(feature_module, rb_intern("MultiPolygon"));
-
-  // Cache some commonly used names
-  globals->id_cast = rb_intern("cast");
-  globals->id_eql = rb_intern("eql?");
-  globals->id_generate = rb_intern("generate");
-  globals->id_enum_for = rb_intern("enum_for");
-  globals->id_hash = rb_intern("hash");
-  globals->sym_force_new = ID2SYM(rb_intern("force_new"));
-  globals->sym_keep_subtype = ID2SYM(rb_intern("keep_subtype"));
+
 #ifndef RGEO_GEOS_SUPPORTS_SETOUTPUTDIMENSION
-  globals->psych_wkt_generator = Qnil;
-  globals->marshal_wkb_generator = Qnil;
+  psych_wkt_generator = Qnil;
+  rb_gc_register_address(&psych_wkt_generator);
+  marshal_wkb_generator = Qnil;
+  rb_gc_register_address(&marshal_wkb_generator);
```

Comment:
```suggestion
  /* We favor rb_gc_register_address over rb_gc_register_mark_object because the value changes at runtime */
  psych_wkt_generator = Qnil;
  rb_gc_register_address(&psych_wkt_generator);
  marshal_wkb_generator = Qnil;
  rb_gc_register_address(&marshal_wkb_generator);
```

Reply:
yeah that makes sense to add a note. I didn't realize you could mark objects with the `rb_gc_register_address` and `rb_gc_mark` functions. Pretty cool